test/e2e/cypress/integration/hana_cluster_details.js

```diff
@@ -0,0 +1,215 @@
+import { availableHanaCluster } from "../fixtures/hana-cluster-details/available_hana_cluster";
+
+context("HANA database details", () => {
+  before(() => {
+    cy.loadScenario("healthy-27-node-SAP-cluster");
+    cy.login();
+
+    cy.visit(`/clusters/${availableHanaCluster.id}`);
+    cy.url().should("include", `/clusters/${availableHanaCluster.id}`);
+  });
+
+  describe("HANA cluster details should be consistent with the state of the cluster", () => {
+    it(`should have name ${availableHanaCluster.name}`, () => {
+      cy.get("div")
```

Comment:
I get your point but I'm not sure I agree on this one; I do think the placement is something worth testing for.

As the UI gets more features, it's quite possible that we will get `.contains()` to match unwanted elements in the UI (it already happened to me). The possibility of using CSS classes is there indeed and would prevent this, but it feels a bit that we are abusing the classes.

Between: 
 - Matching a label and looking for the next item
 - Using CSS classes to target the element we want to test
 
 I'd find the first one the lesser evil :smile:. More opinions on this?

Reply:
>  I do think the placement is something worth testing for.
>
I see your point, and I believe it is a matter of **value**, **perception of value** and **flexibility**.

Here's my argument.
_Sticking on the example, and porting the reasoning to other stuff._
In the current scenario, from my perspective, checking that there in the page a component with the cluster name is more valuable than checking that it is exactly next another one.

Why do I find it more valuable in the current situation? 
Because that would allow us to experiment on the UI in a more efficient way.
We'd still have the value of a test that makes sure a specific information is in the page, with less of a burden when we move things around.

Certainly depends on cases, this is a general recommendation and heuristic.


> It feels a bit that we are abusing the classes.
>
What do you mean? Why do you see it as abusing classes? 
is it because they're not tied to actual styles? :sweat_smile: 
If this is the reason I'd say that:
1. I see that as an acceptable tradeoff considering the previous reasons
2. The alternative would be using [html data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), which I see more expensive

> I'd find the first one the lesser evil smile
>
Why? Could you elaborate a bit more?